File: 4_least_common_multiple/lcm.cpp
/*
 * Author: Jehadur Rahman
 * Organization: CyArm
 * GitHub: https://github.com/JehadurRE
 * Website: jehadurre.me
 * Date: 2025-07-10 09:43 (Thu)
 */

#include <iostream>
#include<cassert>
inline long long lcm_naive(int a, int b) {
  for (long l = 1; l <= (long long) a * b; ++l)
    if (l % a == 0 && l % b == 0)
      return l;

  return (long long) a * b;
}


inline long long gcd (long long a , long long b)
{
  while (b !=0)
  {
    long long temp = b ;
    b = a % b;
    a = temp;
  }
  return a;
}

inline long long lcm_fast(int a , int b )
{
  return (long long )(a / gcd(a, b))*b;
}

void test_solution()
{
  for (int a = 1; a < 20; ++a)
  {
    for (int b = 1; b < 20; ++b)
    {
      assert(lcm_fast(a, b) == lcm_naive(a, b));
    }
  }
  std::cout << "All tests passed!\n";
}
int main() {
  // test_solution();
  int a, b;
  std::cin >> a >> b;
  std::cout << lcm_fast(a, b) << std::endl;
  return 0;
}

File: 4_least_common_multiple/lcm.py
"""
Author: Jehadur Rahman
Organization: CyArm
GitHub: https://github.com/JehadurRE
Website: jehadurre.me
Date: 2025-07-10 09:43 (Thu)
"""

def lcm_naive(a, b):
    for l in range(1, a * b + 1):
        if l % a == 0 and l % b == 0:
            return l
    assert False

def gcd(a, b):
    while b != 0:
        temp = b
        b = a % b
        a = temp
    return a

def lcm_fast(a, b):
    return (a // gcd(a, b)) * b

def test_solution():
    for a in range(1, 20):
        for b in range(1, 20):
            assert lcm_fast(a, b) == lcm_naive(a, b)
    print("All tests passed!")

if __name__ == '__main__':
    # test_solution()
    a, b = map(int, input().split())
    print(lcm_fast(a, b))

